pkg/endpointmanager/manager.go

```diff
@@ -27,145 +27,100 @@ import (
 var (
 	log = common.DefaultLogger
 
-	// Mutex protects Endpoints and endpointsAux
-	//
-	// Warning: This lock may not be taken while an individual endpoint
-	// lock is being held. If you require to hold both, then the global
-	// endpointmanager lock must always be acquired first.
-	Mutex lock.RWMutex
-
-	// Endpoints is the global list of endpoints indexed by ID. Mutex must
+	// mutex protects endpoints and endpointsAux
+	mutex lock.RWMutex
+
+	// endpoints is the global list of endpoints indexed by ID. mutex must
 	// be held to read and write.
-	//
-	// FIXME: This is currently exported, as more code moves from daemon
-	// into pkg/endpoint, we might be able to unexport this
-	Endpoints    = map[uint16]*endpoint.Endpoint{}
+	endpoints    = map[uint16]*endpoint.Endpoint{}
 	endpointsAux = map[string]*endpoint.Endpoint{}
 )
 
-// LookupCiliumIDLocked looks up endpoint by endpoint ID with Mutex held
-func LookupCiliumIDLocked(id uint16) *endpoint.Endpoint {
-	if ep, ok := Endpoints[id]; ok {
-		return ep
-	}
-	return nil
+// Insert inserts the endpoint into the global maps
+func Insert(ep *endpoint.Endpoint) {
+	mutex.Lock()
+	metrics.EndpointCount.Inc()
+	endpoints[ep.ID] = ep
+	updateReferences(ep)
+	mutex.Unlock()
 }
 
-// LookupCiliumID looks up endpoint by endpoint ID
-func LookupCiliumID(id uint16) *endpoint.Endpoint {
-	Mutex.Lock()
-	defer Mutex.Unlock()
-
-	return LookupCiliumIDLocked(id)
-}
+// Lookup looks up the endpoint by prefix id
+func Lookup(id string) (*endpoint.Endpoint, error) {
+	mutex.RLock()
+	defer mutex.RUnlock()
 
-func lookupDockerEndpointLocked(id string) *endpoint.Endpoint {
-	if ep, ok := endpointsAux[endpoint.NewID(endpoint.DockerEndpointPrefix, id)]; ok {
-		return ep
+	prefix, eid, err := endpoint.ParseID(id)
+	if err != nil {
+		return nil, err
 	}
-	return nil
-}
 
-func lookupPodNameLocked(name string) *endpoint.Endpoint {
-	if ep, ok := endpointsAux[endpoint.NewID(endpoint.PodNamePrefix, name)]; ok {
-		return ep
-	}
-	return nil
-}
+	switch prefix {
+	case endpoint.CiliumLocalIdPrefix:
+		n, _ := endpoint.ParseCiliumID(id)
```

Comment:
Can we ignore the error here? n is 0 in those cases, so the cast is safe and the lookup returns the right thing. I guess this is fewer lines of code.

Reply:
Returned error